src/main/java/org/mastodon/blender/csv/ColorSchemeDialog.java

```diff
@@ -0,0 +1,153 @@
+
+	private static final String SEPARATOR = "-----------------";
+
+	/**
+	 * Show a dialog to let the user select a coloring scheme.
+	 * @param projectModel the project model
+	 * @return the name of the selected coloring scheme, or {@code null} if none was selected.
+	 */
+	public static String showDialog( ProjectModel projectModel )
+	{
+		JComboBox< ColorFunction > comboBox = getColorSchemesComboBox( projectModel );
+
+		int result = JOptionPane.showOptionDialog(
+				null,
+				comboBox,
+				"Select coloring scheme",
+				JOptionPane.OK_CANCEL_OPTION,
+				JOptionPane.PLAIN_MESSAGE,
+				null,
+				null,
+				DEFAULT_OPTION
+		);
+
+		if ( result == JOptionPane.OK_OPTION )
+		{
+			ColorFunction selectedItem = ( ColorFunction ) comboBox.getSelectedItem();
+			if ( selectedItem == null )
+				return null;
+			if ( selectedItem.getGroup() == null )
+				return null;
+			return selectedItem.toString();
+		}
+		return null;
+	}
+
+	private static JComboBox< ColorFunction > getColorSchemesComboBox( ProjectModel projectModel )
+	{
+		List< ColorFunction > options = getColorSchemeOptions( projectModel );
+		JComboBox< ColorFunction > comboBox = new JComboBox<>( options.toArray( new ColorFunction[ 0 ] ) );
+		comboBox.setRenderer( new ColorSchemeOptionsRenderer() );
+		return comboBox;
+	}
+
+	static List< ColorFunction > getColorSchemeOptions( ProjectModel projectModel )
+	{
+		List< ColorFunction > colorFunctions = new ArrayList<>();
+		colorFunctions.add( new EmptyColorFunction( DEFAULT_OPTION ) );
+		colorFunctions.add( new EmptyColorFunction( SEPARATOR ) );
+		colorFunctions.add( new EmptyColorFunction( TAGS_HEADER ) );
+		colorFunctions.addAll( GraphToCsvUtils.getTagSetColorFunctions( projectModel ) );
+		colorFunctions.add( new EmptyColorFunction( SEPARATOR ) );
+		colorFunctions.add( new EmptyColorFunction( FEATURE_COLOR_MODES_HEADER ) );
+		colorFunctions.addAll( GraphToCsvUtils.getFeatureColorFunctions( projectModel ) );
+		return colorFunctions;
+	}
+
+	private static class ColorSchemeOptionsRenderer implements ListCellRenderer< ColorFunction >
+	{
+
+		private final DefaultListCellRenderer defaultRenderer = new DefaultListCellRenderer();
+
+		@Override
+		public Component getListCellRendererComponent( final JList< ? extends ColorFunction > list, final ColorFunction colorFunction,
+				final int index, final boolean isSelected, final boolean cellHasFocus )
+		{
+			if ( colorFunction.getGroup() == null )
+			{
+				String name = colorFunction.toString();
+				switch ( name )
+				{
+				case TAGS_HEADER:
+				case FEATURE_COLOR_MODES_HEADER:
+					JLabel label =
+							( JLabel ) defaultRenderer.getListCellRendererComponent( list, colorFunction, index, false, cellHasFocus );
+					label.setFont( label.getFont().deriveFont( Font.BOLD ) );
+					return label;
+				case SEPARATOR:
+					return new JSeparator( SwingConstants.HORIZONTAL );
+				default:
+					return new JLabel( colorFunction.toString() );
```

Comment:
This line `return new JLabel( colorFunction.toString() );` cause the default entry `[no color scheme]` feel not be highlighted when selected. The situation could be improved by simply removing this line.
```suggestion
					// use default renderer
```

Reply:
Good idea. Changed it.